### ISSUE TITLE
no-inferred-empty-object-type incorrectly fails when calling non-generic function

### ISSUE THREAD

**Issue by rrogowski** [palantir/tslint#4364]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export interface Interface {
    doNothing(): void;
}

type InterfaceConstructor = new () => Interface;

export class Implementation implements Interface {
    public doNothing(): void {
        // Nothing to see here
    }
}

function getInterfaceConstructor(): InterfaceConstructor {
    return Implementation;
}

getInterfaceConstructor();
// ~~~~~~~~~~~~~~~~~~~~~~~  [Explicit type parameter needs to be provided to the function call]
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-inferred-empty-object-type": true
  }
}
```

#### Actual behavior

Lint error on `getInterfaceConstuctor` call, claiming to violate the `no-inferred-empty-object-type` rule.

#### Expected behavior

No error. The `no-inferred-empty-object-type` rule is not violated since `getInterfaceConstructor` takes no template parameters.

**Comment by adidahiya** [comment]
@glen-84 which versions are you using?

**Comment by glen-84** [comment]
@adidahiya `tslint@5.13.1`, `typescript@3.3.3333`.

**Comment by leafriend** [comment]
I also failed with `tslint@5.15.0`, `typescript@3.4.1`.

**Comment by danielrentz** [comment]
This fails too:
```
class Test<T = {}> { }
const test = new Test();
```
I think it should pass, as the class explicitly sets `{}` as default type. Is this worth another issue?

**Comment by JoshuaKGoldberg** [comment]
@danielrentz it looks like that would be a separate issue, yes.

**Comment by danielrentz** [comment]
#4770 
Seems these problems cannot be reproduced in the playground BTW...

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
